SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -507,6 +508,22 @@ void USpatialNetDriver::OnGSMQuerySuccess()
 	// If the deployment is now accepting players and we are waiting to spawn. Spawn.
 	if (bWaitingToSpawn && ClientCanSendPlayerSpawnRequests())
 	{
+		// Are we running with the same schema hash as the server?
+#if !(UE_BUILD_SHIPPING)
```

Comment:
@joshuahuburn Is this the right place to do this? I can't see where bWaitingToSpawn is ever set to true except for when we are connecting to a game. Is this the same spawn as a player spawn?

Reply:
@MatthewSandfordImprobable has the best context on where to check for AcceptingPlayers and SessionID, this is basically the same thing. I think so.

Reply:
Looks good to me!

Reply:
I guess this property on the GSM is only set once right? I was just wondering about how we know this value has been updated and isn't the default/ previous value? I believe it is the case that acceptingPlayers and SessionId updates may have been sent to the client however, we might still not have received the updated SchemaHash. If we havn't got that update the GSM->GetSchemaHash is not going to be the same as ClassInfoManager->SchemaDatabase->SchemaDescriptorHash. It might be the case that we need to wait until we see that the GSM->GetSchemaHash is not the "default" value. If the GSM->GetSchemaHash changes multiple times we also need another mechanism.

Reply:
This should actually be ok as you are setting SchemaHash alongside acceptingPlayers and both fields are on the same component.